lib/workers/testbot.ts

```diff
@@ -0,0 +1,278 @@
+
+		return Bluebird.delay(settle);
+	}
+
+	/**
+	 * Connected the SD card interface to the host
+	 *
+	 */
+	private async switchSdToHost(settle: number = 0): Promise<void> {
+		console.log('Switching SD card to host...');
+		await this.resetSdCard();
+		this.board.digitalWrite(GPIO.LED_PIN, this.board.HIGH);
+		this.board.digitalWrite(GPIO.SD_MUX_SEL_PIN, this.board.HIGH);
+
+		return Bluebird.delay(settle);
+	}
+
+	/**
+	 * Power on DUT
+	 */
+
+	private powerOnDUT(): Bluebird<void> {
+		console.log('Switching testbot on...');
+		return this.sendCommand(GPIO.ENABLE_VOUT_SW, 500);
+	}
+
+	/**
+	 * Power off DUT
+	 */
+	private powerOffDUT(): Bluebird<void> {
+		console.log('Switching testbot off...');
+		return this.sendCommand(GPIO.DISABLE_VOUT_SW, 500);
+	}
+
+	/**
+	 * Flash SD card with operating system
+	 */
+	public async flash(url: string): Promise<void> {
+		await this.off();
+
+		return this.criticalSection(async () => {
+			// For linux, udev will provide us with a nice id for the testbot
+			const drive = await getDrive(await this.getDevInterface());
+			const source = await new sdk.sourceDestination.Http(url).getInnerSource();
+			const progressBar: { [key: string]: any } = {
+				flashing: new visuals.Progress('Flashing'),
+				verifying: new visuals.Progress('Validating'),
+			};
+
+			await sdk.multiWrite.pipeSourceToDestinations(
+				source,
+				[drive],
+				(destination, error) => {
+					console.error(error);
+				},
+				(progress: sdk.multiWrite.MultiDestinationProgress) => {
+					progressBar[progress.type].update(progress);
+				},
+				true,
+			);
+		}, arguments);
+	}
+
+	/**
+	 * Get TestBot ready to function
+	 */
+	public ready(): Promise<void> {
+		return this.criticalSection(() => {
+			return new Bluebird((resolve, reject) => {
+				this.board.once('ready', async () => {
+					// Power managment configuration
+					// We set the regulator (DAC_REG) to 5V and start the managment unit (VREG)
+					await this.sendCommand(GPIO.ENABLE_FAULTRST, 1000);
+					this.board.pinMode(GPIO.LED_PIN, this.board.MODES.OUTPUT);
+					await this.sendCommand(GPIO.WRITE_DAC_REG, 1000, 5);
+					await this.sendCommand(GPIO.ENABLE_VREG, 1000);
+
+					// SD card managment configuration
+					// We enable the SD/USB multiplexers and leave them disconnected
+					this.board.pinMode(GPIO.SD_MUX_SEL_PIN, this.board.MODES.OUTPUT);
+					this.board.digitalWrite(GPIO.SD_MUX_SEL_PIN, this.board.LOW);
+					this.board.pinMode(GPIO.USB_MUX_SEL_PIN, this.board.MODES.OUTPUT);
+					this.board.digitalWrite(GPIO.USB_MUX_SEL_PIN, this.board.LOW);
+
+					await Bluebird.delay(1000);
+					console.log('Worker ready');
+
+					resolve();
+				});
+				this.board.once('error', reject);
+			});
+		}, arguments);
+	}
+
+	/**
+	 * Turn on DUT
+	 */
+	public on(): Promise<void> {
+		return this.criticalSection(async () => {
+			await this.switchSdToDUT(5000);
+			await this.powerOnDUT();
+
+			this.disconnect = this.disconnect.bind(this);
```

Comment:
This needs a comment. I think it would make more sense to store the result of this as an instance var, rather than overwriting `this.disconnect`.